code/notebooks/04 3factor_2levels_interaction_effect.ipynb

```diff
@@ -228,6 +230,7 @@
     "  effect_lc=0\n",
     "  effect_hc=0\n",
     "  effect=0\n",
+       x= len(op)/2,
```

Comment:
This looks good. Can you have one same (descriptive) variable name (something like `num_values` should do) for all three functions instead of `l`, `n`, `x`